src/test/java/com/epam/tc/hw2/ex2/Ex2.java

```diff
@@ -0,0 +1,168 @@
+        difElements.click();
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(By.className("uui-main-container")));
+
+        softly.assertThat(driver.getCurrentUrl()).as("Incorrect page opened").isEqualTo("https://jdi-testing.github.io/jdi-light/different-elements.html");
+
+        //6. Select checkboxes Water, Wind
+        WebElement checkboxWater = driver.findElement(By.xpath("//*[contains(text()[normalize-space(.)],'Water')]"));
+        checkboxWater.click();
+        WebElement checkboxWind = driver.findElement(By.xpath("//*[contains(text()[normalize-space(.)],'Wind')]"));
+        checkboxWind.click();
+
+        softly.assertThat(checkboxWater.isEnabled()).as("Check box 'Water' was not enabled").isEqualTo(true);
+        softly.assertThat(checkboxWind.isEnabled()).as("Check box 'Wind' was not enabled").isEqualTo(true);
+
+        //7. Select radio Selen
+        WebElement radioBtnSelen = driver.findElement(By.xpath("//*[contains(text()[normalize-space(.)],'Selen')]"));
+        radioBtnSelen.click();
+
+        softly.assertThat(radioBtnSelen.isEnabled()).as("Incorrect radio button state").isEqualTo(true);
+
+        //8. Select in dropdown Yellow
+        WebElement dropDownColors = driver.findElement(By.className("colors"));
+        dropDownColors.click();
+        WebElement dropDownYellow = driver.findElement(By.xpath("//option[text()='Yellow']"));
+        dropDownYellow.click();
+
+        softly.assertThat(dropDownYellow.isSelected()).as("Incorrect drop down state").isEqualTo(true);
+
+        //9. Assert that
+        driver.navigate().refresh(); //clean all elements state
+        //• for each checkbox there is an individual log row and value is corresponded to the status of checkbox
+        List<WebElement> checkboxes = driver.findElements(By.cssSelector("input[type='checkbox']"));
+        int checkboxesNumber = checkboxes.size();
+        for (int i = 0; i < checkboxesNumber; i++) {
+            checkboxes.get(i).click();
+            String checkboxName = checkboxes.get(i).getText();
+            WebElement log = driver
+                    .findElement(By.cssSelector(".logs :first-child"));
+            System.out.println(log.getText());
+            softly.assertThat(log.getText())
+                    .as("Incorrect log text")
+                    .endsWith(checkboxName + ": condition changed to " + checkboxes.get(i).isSelected());
+
+            checkboxes.get(i).click();
+            WebElement log2 = driver.findElement(By.cssSelector(".logs :first-child"));
+            System.out.println(log2.getText());
+            softly.assertThat(log2.getText())
+                    .as("Incorrect log text")
+                    .endsWith(checkboxName + ": condition changed to " + checkboxes.get(i).isSelected());
+        }
```

Comment:
If you use For-Each Loop there will be less code(without `get(i)`)

Reply:
Also, you can fix it in other places

Reply:
Fixed accross the code